experimental/segmenter/src/lstm.rs

```diff
@@ -87,9 +86,9 @@ impl<'l> LstmSegmenter<'l> {
     pub fn try_new_unstable(
         payload: &'l DataPayload<LstmDataV1Marker>,
         grapheme: Option<&'l RuleBreakDataV1<'l>>,
-    ) -> Result<Self, SegmenterError> {
+    ) -> Option<Self> {
```

Comment:
This is a private type and the error is always being discarded.

Reply:
Depending on the LstmData, we might not need the grapheme though.

Reply:
I don't think it makes sense to create error variants that never reach the user (and are ignored even internally).

Reply:
I'm not strong enough convinced by my position to push back because this is an internal type. My only concern vector is if such type were to ever become public I'd prefer the `try_new_*` to use Result.
Would you be open to rename it? `try_` in Rust stdlib always comes with `Result`. `maybe_new_unstable`?

Reply:
`Result<Self, ()>`?

Reply:
I'll back up the position of empty internal structs as a useful abstraction for internal error types that don't reach the user. I've used them before in ICU4X code.

Reply:
I'm using `()` now as I don't want to name anything.

I've collapsed the `LstmSegmenter` and the `Lstm` types into one, so there's only one of those constructors now.